Fixes two high-priority bugs 1601 and 1549

liach opened the pull request:
**The Issue**
 - #1601 #1549
 - A few other associated bugs identified (see message of the last commit)
 
**The Proposal**
 - Refactors to fix logical loopholes.

**Possible Side Effects**
 - The fluid manipulators have been tested and should be fine.
 - Some code review should be helpful.
 
**Alternatives**
 - None.

liach committed:
Fix #1601

Signed-off-by: liach <liach@users.noreply.github.com>

liach committed:
Merge branch 'mc-1.12.2' of https://github.com/Railcraft/Railcraft into fix/major-bugs

liach committed:
Some code cleanup

Signed-off-by: liach <liach@users.noreply.github.com>

liach committed:
Fixes #1549 fluid loader/unloader loading fluid

The facing was wrong and the inventory handling has some issues as well.
Yeah, I mean that ! in the IExtInvSlot
Make fluid unloader render its fluid with light.
Simplify gui opening code
Style fixes and annotations
Fixes stack filter for fluid container items (a different cap)
Changed log4j settings to show which logger outputs a log

Signed-off-by: liach <liach@users.noreply.github.com>

liach committed:
Fixes #1594

Signed-off-by: liach <liach@users.noreply.github.com>

CovertJaguar commented on src/main/java/mods/railcraft/common/blocks/machine/manipulator/TileFluidLoader.java:
```diff
@@ -54,7 +52,7 @@ public ManipulatorVariant getMachineType() {
 
     @Override
-    public EnumFacing getFacing() {
+    public EnumFacing getDefaultFacing() {
```
Probably shouldn't change this, fluid loaders can't rotate.

At least point getFacing() at getDefaultFacing().

liach replied:
many places access the facing field instead of calling this method. as a result, it's safer to initialize the facing field to default facing and limit rotations instead.

CovertJaguar replied:
Had to fix this to make it compile against my changes to ITileRotate.

CovertJaguar commented on src/main/java/mods/railcraft/common/blocks/machine/manipulator/TileFluidManipulator.java:
```diff
@@ -98,6 +99,7 @@ public boolean blockActivated(EntityPlayer player, EnumHand hand, EnumFacing sid
 
     @Override
     public boolean canHandleCart(EntityMinecart cart) {
+        // TODO opposite facing?
```
Probably.

liach committed:
Fix train dispenser crash #1592
Also makes the cart dispenser no longer consume the cart in its recipe and made it shapeless
Fix shared logic cart killing

Signed-off-by: liach <liach@users.noreply.github.com>

CovertJaguar commented on src/main/java/mods/railcraft/common/blocks/machine/manipulator/TileFluidManipulator.java:
```diff
@@ -113,11 +115,20 @@ protected void upkeep() {
+            FluidStack filter = getFilterFluid();
+            boolean b3 = filter == null;
+            boolean b4 = FluidItemHelper.containsFluid(stack, filter);
+            return b1 && (b2 || b3 || b4);
```
This is less efficient because it looses the short circuit property of Boolean logic.

CovertJaguar replied:
Much easier to read though.

liach replied:
I did that change during testing. Will revert.

CovertJaguar commented on src/main/java/mods/railcraft/common/fluids/tanks/StandardTank.java:
```diff
@@ -105,8 +106,11 @@ public boolean canFillFluidType(FluidStack fluid) {
     }
 
     public boolean matchesFilter(@Nullable FluidStack fluidStack) {
+        if (fluidStack == null) {
```
We actually need to be able to set the tank contents to null.

CovertJaguar replied:
Fixed in merge.

CovertJaguar committed:
Merge branch 'mc-1.12.2' into fix/major-bugs

CovertJaguar committed:
Fix build failure.

CovertJaguar committed:
Fix it the right way this time.

liach committed:
Details.

Signed-off-by: liach <liach@users.noreply.github.com>

CovertJaguar committed:
Convert Rock Crusher to logic framework.

CovertJaguar committed:
Merge remote-tracking branch 'public/fix/major-bugs' into fix/major-bugs
